mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -372,11 +375,12 @@ def _azure_upload_file(self, credentials, local_file, artifact_file_path, get_cr
         """
         try:
             headers = self._extract_headers_from_credentials(credentials.headers)
-            futures = {}
             num_chunks = _compute_num_chunks(local_file, MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.get())
-            for index in range(num_chunks):
-                start_byte = index * MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.get()
-                future = self.chunk_thread_pool.submit(
+            chunk_size = MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.get()
+
+            def job(index):
+                start_byte = index * chunk_size
+                self._azure_upload_chunk(
                     self._azure_upload_chunk,
```

Comment:
this recursive function reference needs to be removed, right?

Reply:
thanks for the catch! must be removed.